protocols/interfaces.go

```diff
@@ -48,9 +48,9 @@ type AdjudicationStatus struct {
 
 // ObjectiveEvent holds information used to update an Objective. Some fields may be nil.
 type ObjectiveEvent struct {
-	ObjectiveId     ObjectiveId
-	SignedStates    []state.SignedState
-	SignedProposals []consensus_channel.SignedProposal
+	ObjectiveId    ObjectiveId
+	SignedState    state.SignedState
```

Comment:
Do we have any constraints on whether an `ObjectiveEvent` can contain both a `SignedProposal` and a `SignedState`?

Reply:
This is checked by the engine when processing messages. msg.SignedStates and msg.Proposals run though [`msg.Type()`](https://github.com/statechannels/go-nitro/blob/9b3db3d9e199f7a8d73c777713ea5a74cb79bb7a/protocols/messages.go#L44) to create the events from the message payloads.

Reply:
Yup, due to our messaging format we expect `ObjectiveEvent` to contain either a `SignedState` or `SignedProposal`.

However it'd be nice if `Update` wasn't implicitly coupled to the message format. If we rely on the expectation that an `ObjectiveEvent` is either a `SignedProposal` or a `SignedState` I think we should make that expectation explicit for the `Update` API. We could this by introducing a `Type` function similar to [messages](https://github.com/statechannels/go-nitro/blob/9b3db3d9e199f7a8d73c777713ea5a74cb79bb7a/protocols/messages.go#L44) but this would add more boilerplate.